docs/services-web-app.md

```diff
@@ -12,3 +29,255 @@ access to private repos and increases API quotas. To set up a personal access to
+instead of the supplied content in the case that the `isAuthorized` prop is set to false
+
+```
+import RequireAuth from '@/components/auth/require-auth'
+import FourOhFour from '@/pages/404'
+...
+const ProtectedPage = (props) => {
+  return (
+    // will return 404 page if isAuthorized is set to false
+    <RequireAuth fallback={FourOhFour} isAuthorized={props.isAuthorized}>
+      <div>User: {JSON.stringify(props?.user ?? {})}</div>
+    </RequireAuth>
+  )
+}
+```
+
+For a working example of a protected server side rendered pages,
+[run the app securely](#running-app-securely) and visit:
+
+- [https://localhost/samples/protectedPageWithSSR](https://localhost/samples/protectedPageWithSSR)
+- [https://localhost/samples/protectedPageWithSSRDynamicAuth](https://localhost/samples/protectedPageWithSSRDynamicAuth)
+- [https://localhost/samples/protectedPageWithSSRDynamicAuth?host=github.com](https://localhost/samples/protectedPageWithSSRDynamicAuth?host=github.com)
+- [https://localhost/samples/protectedPageWithSSRDynamicAuth?host=github.ibm.com](https://localhost/samples/protectedPageWithSSRDynamicAuth?host=github.ibm.com)
+- [https://localhost/samples/protectedPageWithSSRDynamicAuth?host=github.ibm.com&repo=internal-stuff](https://localhost/samples/protectedPageWithSSRDynamicAuth?host=github.ibm.com&repo=internal-stuff)
+- [https://localhost/samples/protectedPageWithSSRDynamicAuth?host=github.ibm.com&repo=private-stuff](https://localhost/samples/protectedPageWithSSRDynamicAuth?host=github.ibm.com&repo=private-stuff)
+
+### Protecting Static Pages
+
+There is no reusable strategy to guarantee authentication before accessing Statically Generated
+Pages; Each page is responisble for handling authentication and authorization at a component-level
+(e.g., fetching user , waiting for user to load and then displaying content). See "Authenticating
+Statically Generated Pages" section in
+[NextJs Authentication Docs](https://nextjs.org/docs/authentication) You may make use of the
+`useAuth()` hook and `RequireAuth` component in your implementations:
+
+```
+  import { useAuth } from 'contexts/auth'
+  import { useEffect } from 'react'
+
+  import RequireAuth from '@/components/auth/require-auth'
+
+  import FourOhFour from '../404'
+
+  const ProtectedStaticPage = () => {
+    const { isAuthenticated, loading, user } = useAuth()
+
+    useEffect(() => {
+      if (!loading && isAuthenticated) {
+        // fetch protected data here
+      }
+    }, [loading, isAuthenticated])
+
+    return loading ? null : (
+      // show page if user is authenticated and email ends with ibm.com, else show 404 page
+      <RequireAuth
+        fallback={FourOhFour}
+        isAuthorized={isAuthenticated && user?.email?.endsWith('ibm.com')}
+      >
+        <>
+          // Your Page Content Here
+        </>
+      </RequireAuth>
+    )
+  }
+
+  // do NOT fetch protected data here
```

Comment:
You can even go a step further here and mention that the reason you can't fetch protected data here is because there's no logged in user available at "build time" when these pages are generated; they're performed automatically by the server at the server's discretion.